website/server/libs/user/index.js

```diff
@@ -126,39 +126,34 @@ export async function update (req, res, { isV3 = false }) {
       _.set(user, key, val);
     } else if (key === 'tags') {
       if (!Array.isArray(val)) throw new BadRequest('mustBeArray');
-
-      const removedTagsIds = [];
-
-      const oldTags = [];
-
-      // Keep challenge and group tags
-      user.tags.forEach(t => {
-        if (t.group) {
-          oldTags.push(t);
-        } else {
-          removedTagsIds.push(t.id);
+      const requestTags = val;
+      const currentTags = user.tags;
+      const currentGroupPlanTags = currentTags.filter(tag => {
+        if (!tag.group) {
+          return false;
         }
-      });
 
-      user.tags = oldTags;
-
-      val.forEach(t => {
-        const oldI = removedTagsIds.findIndex(id => id === t.id);
-        if (oldI > -1) {
-          removedTagsIds.splice(oldI, 1);
+        if (user.party._id && user.party._id === tag.group) {
```

Comment:
this seems to be only checking the user's party but you can also be part of a group that is not your party (see `user.groups`)

Reply:
Do you mean user.guilds ? If yes I check just after at line 140.

Reply:
sorry I missed that